src/terminal/adapter/adaptDispatchGraphics.cpp

```diff
@@ -259,6 +259,38 @@ bool AdaptDispatch::SetGraphicsRendition(const VTParameters options)
     return true;
 }
 
+// Routine Description:
+// - DECSCA - Modifies the character protection attribute. This operation was
+//   originally intended to support a range of logical character attributes,
+//   but the protected attribute was the only one ever implemented.
+// Arguments:
+// - options - An array of options that will be applied in order.
+// Return Value:
+// - True.
+bool AdaptDispatch::SetCharacterProtectionAttribute(const VTParameters options)
+{
+    auto& textBuffer = _api.GetTextBuffer();
+    auto attr = textBuffer.GetCurrentAttributes();
+    for (size_t i = 0; i < options.size(); i++)

```

Comment:
Out of personal curiosity, when looking at the spec at https://vt100.net/docs/vt510-rm/DECSCA.html it looks like DECSCA accepts one single numeric parameter but you iterate over every one (from zero upwards?). Every subsequent parameter will override its predecessor. I can't find that in the linked spec. Why?

Reply:
@christianparpart It's documented in DEC STD-070 on page 5-166 (I can get there directly in my copy of the PDF by selecting page 421). The sequence was designed to be extendable with additional attribute types similar to `SGR`. So just like you can have `\e[1;22m` turning bold on and off again, you can have `\e[1;2"q` turning the protected attribute on and off again.

And just like `\e[0m` turns off all rendition attributes, `\e[0"q` was actually intended to turn off all logical attributes. It's just that in this case that's effectively the same as `\e[2q`, because they never got around to defining any other logical attributes (at least as far as I'm aware).

Reply:
Thanks for that clarification and reference.